File: 2475-number-of-unequal-triplets-in-array/2475-number-of-unequal-triplets-in-array.java
class Solution {
    public int unequalTriplets(int[] A) {
        int trips = 0, pairs = 0, count[] = new int[1001];
        for (int i = 0; i < A.length; ++i) {
            trips += pairs - count[A[i]] * (i - count[A[i]]);
            pairs += i - count[A[i]];
            count[A[i]] += 1;
        }
        return trips;
    }
}

File: 2475-number-of-unequal-triplets-in-array/2475-number-of-unequal-triplets-in-array.py
class Solution:
    def unequalTriplets(self, nums: List[int]) -> int:
        c = Counter(nums)
        #print(c)
        ans = 0
        left = 0
        right = len(nums)
        for _, val in c.items():
            right -= val
            ans+= left*val*right
            left+=val
        return ans